log4j-redis/src/main/java/org/apache/logging/log4j/redis/appender/RedisAppender.java

```diff
@@ -0,0 +1,243 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+
+package org.apache.logging.log4j.redis.appender;
+
+import org.apache.logging.log4j.core.*;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.appender.AppenderLoggingException;
+import org.apache.logging.log4j.core.config.Node;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginBuilderAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginBuilderFactory;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.validation.constraints.Required;
+import org.apache.logging.log4j.core.net.ssl.SslConfiguration;
+import org.apache.logging.log4j.spi.AbstractLogger;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sends log events to a Redis Queue. All logs are appended to Redis lists via the RPUSH command.
+ */
+@Plugin(name = "Redis", category = Node.CATEGORY, elementType = Appender.ELEMENT_TYPE, printObject = true)
+public final class RedisAppender extends AbstractAppender {
+
+    private final RedisManager manager;
+    private final LinkedBlockingQueue<String> logQueue;
+    private final boolean immediateFlush;
+    private final int queueCapacity;
+
+    private RedisAppender(final String name, final Layout<? extends Serializable> layout, final Filter filter,
+                          final boolean ignoreExceptions, boolean immediateFlush, final int queueCapacity, final RedisManager manager) {
+        super(name, filter, layout, ignoreExceptions);
+        this.manager = Objects.requireNonNull(manager, "Redis Manager");
+        this.immediateFlush = immediateFlush;
+        this.queueCapacity = queueCapacity;
+        this.logQueue = new LinkedBlockingQueue<>(queueCapacity);
+    }
+
+    /**
+     * Builds RedisAppender instances.
+     * @param <B> The type to build
+     */
+    public static class Builder<B extends Builder<B>> extends AbstractAppender.Builder<B>
+            implements org.apache.logging.log4j.core.util.Builder<RedisAppender> {
+
+        private final String KEY_SEPARATOR = ",";
+
+        @PluginBuilderAttribute("host")
+        @Required(message = "No Redis hostname provided")
+        private String host;
+
+        @PluginBuilderAttribute("keys")
+        private String keys = "";
+
+        @PluginBuilderAttribute("port")
+        private int port = 6379;
```

Comment:
Can default values be picked up from Jedis?

Reply:
Jedis doesn't expose the default value for port nor offer good constructors to exclude it (i.e. you can't exclude port and provide an SSLConfig). Redis is unlikely to change their default port, though, so this seems pretty safe.

Reply:
A minor improvement might be to refactor all of these initialization values into constants; I'll leave that up to you. But which ever way it goes, it would be nice to Javadoc the port and say what you provided in your comment. That would help for maintenance I think. Thank you!